tasks/ssh.yml

```diff
@@ -1,15 +1,18 @@
 ---
-- stat: path="{{ kdump_sshkey }}"
+- name: Check whether key exists
+  stat: path="{{ kdump_sshkey }}"
   register: sshkey_stats
 
-- command: "/usr/bin/ssh-keygen -t rsa -f {{ kdump_sshkey }} -N '' "
-  when: sshkey_stats.stat.exists == False
+- name: Create key
+  command: "/usr/bin/ssh-keygen -t rsa -f {{ kdump_sshkey }} -N '' "
+  when: not sshkey_stats.stat.exists
 
-- name: cat file to register
-  shell: cat {{ kdump_sshkey }}.pub
+- name: Cat file to register
+  command: cat {{ kdump_sshkey }}.pub
   register: keydata
+  changed_when: False
```

Comment:
and this?
(Of course, problems here will not be caught by Molecule, as it is used only by the default test, which skips this task file.)

Reply:
Both tasks are skipped in check mode and thus last task 'Add key to authorized keys' fails with error:
FAILED! => {"msg": "The task includes an option with an undefined variable. The error was: 'dict object' has no attribute 'stdout'\n\nThe error appears to have been in '/home/vcrhonek/work/packages/rhel-system-roles/kdump/tasks/ssh.yml': line 15, column 3, but may\nbe elsewhere in the file depending on the exact syntax problem.\n\nThe offending line appears to be:\n\n\n- name: Add key to authorized keys\n  ^ here\n"}

Both tasks need to have check_mode: False option set.

Reply:
"Both tasks" means here "Create key" and "Cat file to register" and this is not what I intended. "Create key" should IMO not have check_mode: False, because it changes the managed host, and one should not do that in check mode.

Reply:
We should fix that later:
https://github.com/linux-system-roles/kdump/issues/25